cmd/juju/cloud/showcredential_test.go

```diff
@@ -0,0 +1,195 @@
+	s.api.CheckCallNames(c, "BestAPIVersion", "CredentialContents", "Close")
+}
+
+func (s *ShowCredentialSuite) TestShowCredentialNone(c *gc.C) {
+	s.api.contents = []params.CredentialContentResult{}
+	cmd := cloud.NewShowCredentialCommandForTest(s.api)
+	ctx, err := cmdtesting.RunCommand(c, cmd)
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(cmdtesting.Stderr(ctx), gc.Equals, "No credential to display\n")
+	c.Assert(cmdtesting.Stdout(ctx), gc.Equals, ``)
+	s.api.CheckCallNames(c, "BestAPIVersion", "CredentialContents", "Close")
+}
+
+func (s *ShowCredentialSuite) TestShowCredentialOne(c *gc.C) {
+	s.api.contents = []params.CredentialContentResult{
+		{
+			Result: &params.ControllerCredentialInfo{
+				Content: params.CredentialContent{
+					Cloud:      "cloud-name",
+					Name:       "credential-name",
+					AuthType:   "userpass",
+					Attributes: map[string]string{"username": "fred"},
+					Secrets:    map[string]string{"password": "sekret"},
+				},
+				Models: []params.ModelAccess{
+					{Model: "abcmodel", Access: "admin"},
+					{Model: "xyzmodel", Access: "read"},
+					{Model: "no-access-model"},
+				},
+			},
+		},
+	}
+	cmd := cloud.NewShowCredentialCommandForTest(s.api)
+	ctx, err := cmdtesting.RunCommand(c, cmd)
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(cmdtesting.Stderr(ctx), gc.Equals, ``)
+	c.Assert(cmdtesting.Stdout(ctx), gc.Equals, `
+credential-name:
+  auth-type: userpass
+  cloud: cloud-name
+  attributes:
+    username: fred
+  secrets:
+    password: sekret
+  models:
+    abcmodel: admin
+    no-access-model: '-'
+    xyzmodel: read
+`[1:])
+	s.api.CheckCallNames(c, "BestAPIVersion", "CredentialContents", "Close")
+}
+
+func (s *ShowCredentialSuite) TestShowCredentialMany(c *gc.C) {
+	s.api.contents = []params.CredentialContentResult{
+		{
+			Result: &params.ControllerCredentialInfo{
+				Content: params.CredentialContent{
+					Cloud:      "cloud-name",
+					Name:       "one",
+					AuthType:   "userpass",
+					Attributes: map[string]string{"username": "fred"},
+					// Don't have secrets here.
+				},
+				// Don't have models here.
+			},
+		}, {
+			Error: common.ServerError(errors.New("boom")),
+		}, {
+			Result: &params.ControllerCredentialInfo{
+				Content: params.CredentialContent{
+					Cloud:    "cloud-name",
+					Name:     "two",
+					AuthType: "userpass",
+					Attributes: map[string]string{
+						"username":  "fred",
+						"something": "visible-attr",
+					},
+					Secrets: map[string]string{
+						"password": "sekret",
+						"hidden":   "very-very-sekret",
+					},
+				},
+				Models: []params.ModelAccess{
+					{Model: "abcmodel", Access: "admin"},
+					{Model: "xyzmodel", Access: "read"},
+					{Model: "no-access-model"},
+				},
+			},
+		},
+	}
+	cmd := cloud.NewShowCredentialCommandForTest(s.api)
+	ctx, err := cmdtesting.RunCommand(c, cmd)
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(cmdtesting.Stderr(ctx), gc.Equals, "boom\n")
+	c.Assert(cmdtesting.Stdout(ctx), gc.Equals, `
+one:
```

Comment:
I am unsure about the output. I think it needs to be consistent with that of list-credentials. ie at the least, a top level "credentials" field. Also, cloud before auth type. If it were to be totally consistent, cloud would have its own level. Note also for list-credentials, secrets are just attributes. They are either included or not but don't get their special section. Plus we don't have a separate attributes section.
Maybe this was discussed and agreed to? But I would like us to try and be consistent in how we display credential info. IMO.

```yaml
credentials:
  one:
    cloud: cloud-name
    auth-type: userpass
    username: fred
    password: secret
    models:
      abc: admin
  two:
    cloud: cloud-name2
    auth-type: userpass
    username: mary
```
For reference, this is what list-credentials does:
```yaml
credentials:
  aws:
    bob:
      auth-type: access-key
      access-key: key
      secret-key: secret
    down:
      auth-type: userpass
      password: password
      username: user
  azure:
    azhja:
      auth-type: userpass
      application-id: app-id
      application-password: app-secret
      subscription-id: subscription-id
      tenant-id: tenant-id
```

Reply:
The grouping will mean that there will be another level introduced in the output where we hope only one credential is always displayed. Also do not forget that here we'd get a collection with models which gives you another level...

So your example will look like this...

credentials:
  cloud-name:
    one:
      auth-type: userpass
      username: fred
      password: secret
      models:
        abc: admin
  cloud-name2:
    two: 
      auth-type: userpass
      username: mary

Reply:
'''credentials:
  cloud-name:
    one:
      auth-type: userpass
      username: fred
      password: secret
      models:
        abc: admin
  cloud-name2:
    two: 
      auth-type: userpass
      username: mary'''

Reply:
So, the problem with ordered maps and inline in yaml is that "models" will be sorted among the attributes keys. This means that it may not be the last element in the output. To guarantee that it is last, I will have to change the output slightly:

```
credentials:
  cloud-name:
    one:
      content:
        auth-type: userpass
        username: fred
        password: secret
      models:
        abc: admin
  cloud-name2:
    two: 
      content:
        auth-type: userpass
        username: mary
      models:
```

Note that if a credential is not used in any models, I will still display the "models" key.